src/main/java/com/ita/if103java/ims/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+import com.ita.if103java.ims.service.MailService;
+import com.ita.if103java.ims.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private UserDao userDao;
+    private UserDtoMapper userDtoMapper;
+    private MailService mailService;
+
+    @Autowired
+    public UserServiceImpl(UserDao userDao, UserDtoMapper userDtoMapper, MailService mailService) {
+        this.userDao = userDao;
+        this.userDtoMapper = userDtoMapper;
+        this.mailService = mailService;
+    }
+
+    @Override
+    public UserDto create(UserDto userDto) {
+        User createdUser = userDao.create(userDtoMapper.convertUserDtoToUser(userDto));
+        String activationURL = "http://localhost:8080/dashboards/" + createdUser.getEmailUUID();
+        sendActivationMessage(userDto, activationURL);
+        return userDtoMapper.convertUserToUserDto(createdUser);
+    }
+
+    @Override
+    public UserDto findById(Long id) {
+        return userDtoMapper.convertUserToUserDto(userDao.findById(id));
+    }
+
+    @Override
+    public UserDto findByAccountId(Long accountID) {
+        return userDtoMapper.convertUserToUserDto(userDao.findByAccountId(accountID));
+    }
+
+    @Override
+    public UserDto update(UserDto userDto) {
+        User updatedUser = userDtoMapper.convertUserDtoToUser(userDto);
+        //Activating status can't be changed in this way
+        User DBUser = userDao.findById(updatedUser.getId());
+        updatedUser.setActive(DBUser.isActive());
+        return userDtoMapper.convertUserToUserDto(userDao.update(updatedUser));
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return userDao.delete(id);
+    }
+
+    @Override
+    public List<UserDto> findAll() {
+        return userDtoMapper.convertToUserDtoList(userDao.findAll());
+    }
+
+    @Override
+    public UserDto findByEmail(String email) {
+        return userDtoMapper.convertUserToUserDto(userDao.findByEmail(email));
+    }
+
+    @Override
+    public boolean updatePassword(Long id, String newPassword) {
+        return userDao.updatePassword(id, newPassword);
+    }
+
+    @Override
+    public boolean activateUser(String emailUUId) {
+        User activatedUser = userDao.findByEmailUUID(emailUUId);
+        ZonedDateTime createdDateTime = activatedUser.getCreatedDate();
+        ZonedDateTime currrentDateTime = ZonedDateTime.now(ZoneId.systemDefault());
+
+        Duration duration = Duration.between(createdDateTime, currrentDateTime);
+        long diffMinutes = Math.abs(duration.toMinutes());
+        if (diffMinutes <= 24 * 60) {
+            activatedUser.setActive(true);
+            userDao.update(activatedUser);
+            return true;
+        }
+        return false;
```

Comment:
I think in this case additionally we need to hard delete user from the DB in order to allow to register again with the same email, otherwise we will block user's email for further registrations in our system